var/spack/repos/builtin/packages/chapel/package.py

```diff
@@ -3,24 +3,385 @@
 #
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
+import os
+
 from spack.package import *
+from spack.util.environment import path_put_first, set_env
 
 
 class Chapel(AutotoolsPackage):
     """Chapel is a modern programming language that is parallel, productive,
     portable, scalable and open-source."""
 
     homepage = "https://chapel-lang.org/"
-    url = "https://github.com/chapel-lang/chapel/releases/download/1.24.1/chapel-1.24.1.tar.gz"
 
+    url = "https://github.com/chapel-lang/chapel/archive/refs/tags/1.33.0.tar.gz"
+    git = "https://github.com/chapel-lang/chapel.git"
+
+    # A list of GitHub accounts to notify when the package is updated.
+    maintainers("arezaii")
+
+    # See https://spdx.org/licenses/ for a list.
     license("Apache-2.0")
 
-    version("1.24.1", sha256="f898f266fccaa34d937b38730a361d42efb20753ba43a95e5682816e008ce5e4")
-    version("1.24.0", sha256="77c6087f3e0837268470915f2ad260d49cf7ac4adf16f5b44862ae624c1be801")
-    version("1.23.0", sha256="7ae2c8f17a7b98ac68378e94a842cf16d4ab0bcfeabc0fee5ab4aaa07b205661")
-    version("1.22.1", sha256="8235eb0869c9b04256f2e5ce3ac4f9eff558401582fba0eba05f254449a24989")
-    version("1.22.0", sha256="57ba6ee5dfc36efcd66854ecb4307e1c054700ea201eff73012bd8b4572c2ce6")
-    version("1.21.0", sha256="886f7ba0e0e86c86dba99417e3165f90b1d3eca59c8cd5a7f645ce28cb5d82a0")
-    version("1.20.0", sha256="08bc86df13e4ad56d0447f52628b0f8e36b0476db4e19a90eeb2bd5f260baece")
-    version("1.19.0", sha256="c2b68a20d87cc382c2f73dd1ecc6a4f42fb2f590b0b10fbc577382dd35c9e9bd")
-    version("1.18.0", sha256="68471e1f398b074edcc28cae0be26a481078adc3edea4df663f01c6bd3b6ae0d")
+    version("main", branch="main")
+
+    version("1.33.0", sha256="c7dfe691a043b6a5dcbea6fe7607ca030014f1a8019744c4c99f67caa8829ba3")
+    version("1.32.0", sha256="a359032b4355774e250fb2796887b3bbf58d010c468faba97f7b471bc6bab57d")
+    version("1.31.0", sha256="bf9a63f7e5d1f247e8680c9a07aeb330cbbf199777a282408100a87dda95918f")
+    version("1.30.0", sha256="d7d82f64f405b8c03e2ce6353d16aba5a261d3f0c63dc3bb64ea3841cfa597b9")
+    version(
+        "1.29.0",
+        deprecated=True,
+        sha256="7fcd13db8e27f14d586358d4c2587e43c8f21d408126fa0ca27d1b7067b867c0",
+    )
+    version(
+        "1.28.0",
+        deprecated=True,
+        sha256="321243a91f8f2dfb3b37a714e2d45298e6a967a9a115565f9ad9cc630ff0bd0e",
+    )
+    # Do NOT add older versions here.
+    # Chapel releases over 2 years old are not supported.
+
+    depends_on("doxygen@1.8.17:")
+
+    variant(
+        "llvm",
+        default="unset",
```

Comment:
Doesn't a default of `CHPL_LLVM=unset` mean that it won't build unless you provide one? Would the default being `spack` would be more reasonable here than making it required?

Reply:
I have been wrestling with reasonable defaults for some of these. The pain of specifying spack by default is that it will likely have to build from source. It might end up being the best default we can offer though.